### DISCUSSION TITLE
How do I clone and run the project on my machine?

### DISCUSSION THREAD

**Discussion by Bruno12leonel** [online-ml/river#1183] [Q&A] [answered]
I'm developing an algorithm for dataflow and would like to use a ready-made structure from one of the river algorithms, making some changes. How do I clone the project and run this new code I'm developing on my machine?
Thanks in advance.

**Comment by MaxHalford** [comment, marked as answer]
Hmm I think this is a bit off-topic and pertains more to how to use GitHub and git in general. I suggest following the official docs, starting [here](https://docs.github.com/en/desktop/contributing-and-collaborating-using-github-desktop/adding-and-cloning-repositories/cloning-and-forking-repositories-from-github-desktop).

Actually, are you looking to contribute something? Or is this for personal usage?

**Reply by Bruno12leonel** [reply]
Thanks.
For now it's for personal use but I hope I can contribute.

**Reply by MaxHalford** [reply]
All the models in River inherit from base classes in the base module. You can inherit from them if you want to create new classes.
